### PR TITLE
test: e2e to reveal private key

### PR DESCRIPTION
**Pull request by plasmacorral**
## **Description**

This PR adds e2e regression tests for revealing the private key.  We reveal private keys for both an HD and imported account from both settings and the account menu.  This test also confirms that the private key is not revealed when using an incorrect password.

It was based on some of the work in [10879](https://github.com/MetaMask/metamask-mobile/pull/10879) that reveals SRP.

## **Related issues**

Fixes: [493](https://github.com/MetaMask/accounts-planning/issues/493)

## **Manual testing steps**

Manual testing steps
On local machine run on this branch...
`Android: yarn test:e2e:android:debug:build` then `yarn test:e2e:android:debug:run e2e/specs/accounts/reveal-private-key.spec.js`
iOS: `yarn test:e2e:ios:debug:build` then `yarn test:e2e:ios:debug:run e2e/specs/accounts/reveal-private-key.spec.js`

pr_regression_e2e_pipeline: [*Pass](https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/65044f6b-20d1-40c3-9411-5956d86e323a) *Other regression test failures are under investigation

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: e2e/selectors/Modals/SrpQuizModal.selectors.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import enContent from '../../../locales/languages/en.json';
+
+export const SrpQuizGetStartedSelectors = {
+  IDs: {
```
Comment (cortisiko) [inline e2e/selectors/Modals/SrpQuizModal.selectors.js:4]: Mind updating the selectors to following our best practices guide? https://github.com/MetaMask/contributor-docs/blob/main/docs/e2e/mobile-e2e-guidelines.md#test-ids

File: e2e/pages/modals/SrpQuizModal.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      ? Matchers.getElementByID(
+          SrpSecurityQuestionOneSelectors.IDs.RIGHT_CONTINUE,
+        )
+      : Matchers.getElementByID(
+          SrpSecurityQuestionTwoSelectors.IDs.RIGHT_CONTINUE,
+        );
+  }
+
+  // Methods for common actions
+  async tapGetStartedDismiss() {
```
Comment (cortisiko) [inline e2e/pages/modals/SrpQuizModal.js:135]: this name is a bit vague. Is it tapping on the dismiss button or is it tapping on get started?

**Review by cortisiko — DISMISSED** [review]
left some small feedback.

File: e2e/pages/modals/SrpQuizModal.js (hunk 2 above)
Comment (plasmacorral) [reply e2e/pages/modals/SrpQuizModal.js:135]: Nice catch, thank you. This is dismissing the get started screen.  I renamed this to `tapGetStartedScreenDismiss`

File: e2e/selectors/Modals/SrpQuizModal.selectors.js (hunk 1 above)
Comment (plasmacorral) [reply e2e/selectors/Modals/SrpQuizModal.selectors.js]: This PR was based on the in-flight work to reveal SRP.  I addressed this suggestion on PR [10754](https://github.com/MetaMask/metamask-mobile/pull/10754) and rebased this PR.  Once I get SRP reveal merged I will rebase on main and take this PR out of draft.  Thank you for taking the time to provide feedback!

File: e2e/specs/accounts/reveal-private-key.spec.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    );
+    await RevealPrivateKey.scrollToDone();
+    await RevealPrivateKey.tapDone();
+  }
+
+  /**
+   * Helper function to handle incorrect password scenario.
+   * @param {string} password - The incorrect password to enter.
+   */
+  async function handleIncorrectPassword(password) {
```
Comment (cortisiko) [inline e2e/specs/accounts/reveal-private-key.spec.js:112]: There is no need to have a function for assertions in the test file itself. Any test action needs to live in the page object itself. The assertions should live within the test.

**Review by owencraston — DISMISSED** [review]

File: e2e/specs/accounts/reveal-private-key.spec.js (hunk 3 above)
Comment (plasmacorral) [reply e2e/specs/accounts/reveal-private-key.spec.js]: removed, thanks!

**Review by owencraston — APPROVED** [review]

**Pull request merged**